File: frontend/src/api/compliance.ts
import api from "./client";

export interface ComplianceItem {
  id: string;
  proposal_id: string;
  item_name: string;
  category: string;
  status: string;
  assigned_to: string | null;
  notes: string | null;
  order_index: number;
  updated_at: string | null;
}

export const complianceApi = {
  list: (proposalId: string) =>
    api.get<ComplianceItem[]>(`/api/proposals/${proposalId}/compliance/`).then(r => r.data),
  create: (proposalId: string, data: Partial<ComplianceItem>) =>
    api.post<ComplianceItem>(`/api/proposals/${proposalId}/compliance/`, data).then(r => r.data),
  init: (proposalId: string) =>
    api.post<ComplianceItem[]>(`/api/proposals/${proposalId}/compliance/init`).then(r => r.data),
  update: (proposalId: string, id: string, data: Partial<ComplianceItem>) =>
    api.patch<ComplianceItem>(`/api/proposals/${proposalId}/compliance/${id}`, data).then(r => r.data),
  delete: (proposalId: string, id: string) =>
    api.delete(`/api/proposals/${proposalId}/compliance/${id}`),
};

File: backend/app/models/compliance.py
import uuid
import enum
from sqlalchemy import Column, String, Integer, Text, DateTime, ForeignKey, func
from sqlalchemy.dialects.postgresql import UUID
from app.db.base import Base


class ComplianceStatus(str, enum.Enum):
    pending = "pending"
    in_progress = "in_progress"
    completed = "completed"
    not_applicable = "not_applicable"


class ComplianceItem(Base):
    __tablename__ = "compliance_items"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    proposal_id = Column(UUID(as_uuid=True), ForeignKey("proposals.id", ondelete="CASCADE"), nullable=False, index=True)
    item_name = Column(String, nullable=False)
    category = Column(String, nullable=False)
    status = Column(String, nullable=False, default="pending")
    assigned_to = Column(String, nullable=True)
    notes = Column(Text, nullable=True)
    order_index = Column(Integer, nullable=False, default=0)
    updated_by = Column(String, nullable=True)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
